code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/DeploymentUnit.java

```diff
@@ -84,6 +84,9 @@ private void createMissingUnitInstances(Map<Long, DeploymentUnitInstanceIdGenera
     }
 
     public boolean isError() {
+        if (!this.isComplete()) {
```

Comment:
Currently we don't keep track of the relationship within the deploymentUnit, so if any part of it is missing, the entire unit has to be recreated.  For example, when the sidekick instance is gone, the living parent is no longer valid as it references non-existing container.